plugin/src/main/java/org/opensearch/sql/plugin/rest/RestPPLQueryAction.java

```diff
@@ -112,43 +111,42 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient nod
             PPLQueryRequestFactory.getPPLRequest(request)
     );
 
-    return channel -> schedule(nodeClient, () ->
-      nodeClient.execute(
-          PPLQueryAction.INSTANCE,
-          transportPPLQueryRequest,
-          new ActionListener<>() {
-            @Override
-            public void onResponse(TransportPPLQueryResponse response) {
-              sendResponse(channel, OK, response.getResult());
-            }
-
-            @Override
-            public void onFailure(Exception e) {
-              if (transportPPLQueryRequest.isExplainRequest()) {
-                LOG.error("Error happened during explain", e);
-                sendResponse(
-                    channel,
-                    INTERNAL_SERVER_ERROR,
-                    "Failed to explain the query due to error: " + e.getMessage());
-              } else if (e instanceof IllegalAccessException) {
-                reportError(channel, e, BAD_REQUEST);
-              } else {
-                LOG.error("Error happened during query handling", e);
-                if (isClientError(e)) {
-                  Metrics.getInstance()
-                      .getNumericalMetric(MetricName.PPL_FAILED_REQ_COUNT_CUS)
-                      .increment();
+    return channel ->
```

Comment:
remove the schedule logic. In OpenSearch, QueryManager will schedule the query in sql-worker thread.